### PR TITLE
Build bowtie on macOS ARM

### PR DESCRIPTION
**Pull request by akikuno**
Describe your pull request here

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by akikuno** [comment]
@BiocondaBot please add labe

**Comment by akikuno** [comment]
@BiocondaBot please add label

File: recipes/bowtie/build.sh [hunk 1]
```diff
@@ -1,5 +1,9 @@
 #!/bin/bash
 
+if [ "$(uname -m)" == "arm64" ]; then
```
Comment (martin-g) [inline recipes/bowtie/build.sh:3]:
```suggestion
set -xe

if [ "$(uname -m)" == "arm64" ]; then
```

File: recipes/bowtie/build.sh [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+    make POPCNT_CAPABILITY=0
```
Comment (martin-g) [inline recipes/bowtie/build.sh:4]:
```suggestion
    make POPCNT_CAPABILITY=0 -j ${CPU_COUNT}
```

File: recipes/bowtie/build.sh [hunk 3]
```diff
@@ -1,5 +1,9 @@
...
+fi
+
 make prefix="${PREFIX}" install
```
Comment (martin-g) [inline recipes/bowtie/build.sh:7]:
```suggestion
make prefix="${PREFIX}"  -j ${CPU_COUNT} install
```

File: recipes/bowtie/build.sh (hunk 2 above)
Comment (corneliusroemer) [inline recipes/bowtie/build.sh]: Maybe a brief comment on why POPCNT_CAPABILITY needs to be disabled for arm64 only would be great!

Comment (akikuno) [reply recipes/bowtie/build.sh]: @corneliusroemer 
I've added a comment about it. Thank you!

Comment (corneliusroemer) [reply recipes/bowtie/build.sh]: I see that debian has the same patch: https://sources.debian.org/patches/bowtie/1.3.0%2Bdfsg1-1/popcnt_capability.patch/ :)

File: recipes/bowtie/build.sh [hunk 4]
```diff
@@ -1,5 +1,15 @@
 #!/bin/bash
 
-make prefix="${PREFIX}" install
+set -xe
+
+# If the environment is arm64 architecture, set POPCNT_CAPABILITY to 0 and run make.
+# This is because the POPCNT instruction is not supported on arm64 architecture.
+# By setting POPCNT_CAPABILITY=0, this instruction is disabled, ensuring build compatibility.
```
Comment (corneliusroemer) [inline recipes/bowtie/build.sh:7]:
```suggestion
# By setting POPCNT_CAPABILITY=0, this instruction is disabled, ensuring build compatibility.
# See https://github.com/BenLangmead/bowtie/blob/master/processor_support.h
```

**Review by martin-g — APPROVED** [review]

**Pull request merged**